-Wstrict-overflow warning from systems/trajectory_optimization/multiple_shooting.h

On Bionic, GCC 7:
```
drake/systems/trajectory_optimization/multiple_shooting.h:109:29: warning: assuming signed overflow does not occur when assuming that (X - c) <= X is always true [-Wstrict-overflow]
     DRAKE_DEMAND(index >= 0 && index < N_);
                  ~~~~~~~~~~~^~~~~~~~~~~~
bazel-out/k8-opt/bin/common/_virtual_includes/essential/drake/common/drake_assert.h:108:26: note: in definition of macro 'DRAKE_DEMAND'
     if (!Trait::Evaluate(condition)) {                                       \
```

--- a/systems/trajectory_optimization/multiple_shooting.h
+++ b/systems/trajectory_optimization/multiple_shooting.h
@@ -106,7 +106,8 @@ class MultipleShooting : public solvers::MathematicalProgram {
   /// index @p index.
   Eigen::VectorBlock<const solvers::VectorXDecisionVariable> input(
       int index) const {
-    DRAKE_DEMAND(index >= 0 && index < N_);
+    DRAKE_DEMAND(index >= 0);
+    DRAKE_DEMAND(index < N_);
     return u_vars_.segment(index * num_inputs_, num_inputs_);
   }
 

